### DISCUSSION TITLE
Get a custom design system with IntelliSense.

### DISCUSSION THREAD

**Discussion by xzfd1010** [tailwindlabs/tailwindcss#4871] [Help]
I want to generate a complete design system that doesn’t extend Tailwind’s defaults, at the same time, I'd like to use
IntelliSense for VS Code to improve development experience.
While using vscode plugins, I don't need the default tailwind classNames like 'row-end-1'. 
![image](https://user-images.githubusercontent.com/15313636/124256576-0c051300-db5e-11eb-9949-ba47e727b8cc.png)
What I need is just the classNames generated by my own config file.
According to the [document](https://tailwindcss.com/docs/presets#disabling-the-default-configuration),  I set `preset` as `[]`, and import my own preset as following: 
## preset.js
```javascript
module.exports = {
    presets: [],
    corePlugins: [
        'textColor',
        'borderRadius',
        'fontWeight',
        'backgroundColor',
        'borderColor',
        'flex'
    ],
    theme: {
        screens: {},
        spacing: {
            'xxs': '2px',
            'xs': '5px',
            's': '7px',
            'm': '10px',
            'l': '15px',
            'xl': '20px',
            'row': '17px'
        },
        borderRadius: {
            's': '2px',
            'm': '4px',
            'l': '8px',
            'xl': '12px',
            'circle': '100%',
        },
        fontWeight: {
            regular: 'normal',
            medium: '500',
        },
        borderColor: {
            DEFAULT: 'rgba(0,0,0,0.03)',
            dark: 'rgba(255,255,255,0.03)',
        },
    },
    variants: {
        textColor: [],
        backgroundColor: [],
        borderColor: [],
        borderRadius: []
    }
  }
```
taiwind.config.js
```
module.exports = {
    presets:[
        require('./preset')
    ],
}
```
When I run `npx tailwind build`, it compiles with error.
![image](https://user-images.githubusercontent.com/15313636/124262124-3bb71980-db64-11eb-81c1-7cf7f9a4f17e.png)
At the same time, the vscode plugin reports the same error.

The reason for the error is that `Object.entries(values)` does not judge if values is an empty object.
 ```javascript
let modifierValues = Object.entries(values);
```

Could you please fix it?

**Comment by olets** [comment]
You can get the result you want by moving the definitions in your presets file to main config file and removing the `presets: []`. Defining `theme.<plugin>` rather than `theme.extend.<plugin>` wipes out the default config for `<plugin>`, and any core plugins not in `corePlugins` are disabled.

This does not directly address the error you're seeing, but it does work around it.

https://play.tailwindcss.com/3H8rZGnLM6

![image](https://user-images.githubusercontent.com/3282350/124321614-b7d83f80-db19-11eb-8f6f-5cf762773aa2.png)

**Reply by xzfd1010** [reply]
Thank you for your attention.
I tried you solution. It works in the browser but it seems not to work in my vscode. The plugin still shows irrelevant classnames as following.
![image](https://user-images.githubusercontent.com/15313636/124581674-e0e33200-de83-11eb-90f2-6a3ec09fd6c4.png)
This is the tailwind config file.
```javascript
module.exports = {
    // change: commented out the following line
    // presets: [],
    corePlugins: [
      'textColor',
      'borderRadius',
      'fontWeight',
      'backgroundColor',
      'borderColor',
      'flex',
    ],
    theme: {
      screens: {},
      spacing: {
        xxs: '2px',
        xs: '5px',
        s: '7px',
        m: '10px',
        l: '15px',
        xl: '20px',
        row: '17px',
      },
      borderRadius: {
        s: '2px',
        m: '4px',
        l: '8px',
        xl: '12px',
        circle: '100%',
      },
      fontWeight: {
        regular: 'normal',
        medium: '500',
      },
      borderColor: {
        DEFAULT: 'rgba(0,0,0,0.03)',
        dark: 'rgba(255,255,255,0.03)',
      },
    },
    variants: {
      textColor: [],
      backgroundColor: [],
      borderColor: [],
      borderRadius: [],
    },
  }
  
```
